src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -357,6 +359,41 @@ class spell_pal_blinding_light : public SpellScript
     }
 };
 
+// 1022 - Blessing of Protection
+class spell_pal_blessing_of_protection : public SpellScript
+{
+    PrepareSpellScript(spell_pal_blessing_of_protection);
+    
+    bool Validate(SpellInfo const* /*SpellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_PALADIN_FORBEARANCE, SPELL_PALADIN_IMMUNE_SHIELD_MARKER });
+    }
+    SpellCastResult Forbearance()
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if(caster->HasAura(SPELL_PALADIN_FORBEARANCE) || caster->HasAura(SPELL_PALADIN_IMMUNE_SHIELD_MARKER))
+                return SPELL_FAILED_TARGET_AURASTATE;
+        }
+        return SPELL_CAST_OK;
+    }
+    
+    void HandleOnHit()
+    {
+        if (Unit* caster = GetCaster())
+        {
+            caster->CastSpell(caster, SPELL_PALADIN_FORBEARANCE, true);
+            caster->CastSpell(caster, SPELL_PALADIN_IMMUNE_SHIELD_MARKER, true);
```

Comment:
i think these auras should be cast on target

Reply:
Hi can you make changes as needed i dont have acces to my pc